### PR TITLE
Update theme colors in new ui

### PR DESCRIPTION
**Pull request by bbovenzi**
Taking some inspiration from the [Chakra-UI 3 announcement](https://www.chakra-ui.com/blog/00-announcing-v3) to improve how we handle colors. Make our own version of semantic colors for our common accent color, blue. Only worry about dark/light mode in our theme and not in components themselves.

We may switch to Chakra UI v3 once its had a few patch releases.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/ui/src/theme.ts [hunk 1]
```diff
@@ -67,8 +72,16 @@ const theme = extendTheme({
...
     colors: {
-      "subtle-bg": { _dark: "gray.900", _light: "blue.50" },
-      "subtle-text": { _dark: "blue.500", _light: "blue.600" },
+      /* eslint-disable perfectionist/sort-objects */
+      contrast: { _dark: "blue.200", _light: "blue.600" },
+      focusRing: "blue.500",
+      fg: { _dark: "blue.600", _light: "blue.400" },
+      emphasized: { _dark: "blue.700", _light: "blue.300" },
+      solid: { _dark: "blue.800", _light: "blue.200" },
+      muted: { _dark: "blue.900", _light: "blue.100" },
```
Comment (ryanahamilton) [inline airflow/ui/src/theme.ts]: Should these token names be more specific, e.g. bgMuted? Id expect this color to be different for something like muted text (e.g. bodyTextMuted).

**Review by jscheffl — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

File: airflow/ui/src/theme.ts (hunk 1 above)
Comment (bbovenzi) [reply airflow/ui/src/theme.ts]: These are the color names in chakra v3. So I wanted to practice using them. Which `fg` and `emphasized` are the colors for foreground elements like text, borders, icons. This is a bit exploratory too. I might need to adjust it later.

**Comment by ryanahamilton** [comment]
In v3 it appears that they are tokens per color palette, so something like
`"blue.muted"` would probably be more inline with how they use them.

**Comment by bbovenzi** [comment]
> In v3 it appears that they are tokens per color palette, so something like `"blue.muted"` would probably be more inline with how they use them.

Good point. Updated.

**Review by ryanahamilton — APPROVED** [review]
Nice!

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**